packages/block-library/src/navigation/style.scss

```diff
@@ -686,10 +684,8 @@ button.wp-block-navigation-item__content {
 		font-size: inherit;
 	}
 
-	&:not(.always-shown) {
-		@include break-small {
-			display: none;
-		}
+	:not(.is-collapsed) > & {
```

Comment:
Again a comment by way of explanation in the code would help here.

Reply:
Done in ad7533893c